google-cloud-firestore/src/test/java/com/google/cloud/firestore/QueryTest.java

```diff
@@ -902,6 +907,72 @@ public void onCompleted() {
     semaphore.acquire();
   }
 
+  @Test
+  public void retriesAfterError() throws Exception {
+    final boolean[] returnError = new boolean[] {true};
+
+    doAnswer(
+            new Answer<RunQueryResponse>() {
+              public RunQueryResponse answer(InvocationOnMock invocation) throws Throwable {
+                if (returnError[0]) {
+                  returnError[0] = false;
+                  return queryResponse(
+                          FirestoreException.serverRejected(
+                              Status.DEADLINE_EXCEEDED, "Simulated test failure"),
```

Comment:
I wonder if it makes sense to add a test to verify the no-retry behavior where it is not suppose to re-try.